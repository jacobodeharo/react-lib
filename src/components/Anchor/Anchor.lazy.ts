const LazyAnchor = () =>
  import('../Anchor/Anchor.styles').then(({ AnchorStyles }) => ({
    default: AnchorStyles,
  }));

export { LazyAnchor };
