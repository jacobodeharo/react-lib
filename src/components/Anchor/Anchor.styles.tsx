import styled from 'styled-components';
import { AnchorProps } from './Anchor.types';

const AnchorStyles = styled.a<AnchorProps>`
  color: var(--c-anchor-text, ${({ theme }) => theme.primaryTextColor});
  transition: color 150ms ease-in-out;
  outline: none;

  :hover {
    color: var(--c-anchor-text-hover, ${({ theme }) => theme.primaryColorDark});
    cursor: pointer;
  }
`;
export { AnchorStyles };
