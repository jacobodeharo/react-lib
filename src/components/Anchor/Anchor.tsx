import React from 'react';
import { AnchorProps } from './Anchor.types';
import styled from 'styled-components';

const Anchor = styled.a<AnchorProps>`
  color: var(--c-text, ${({ theme }) => theme.cText});
  transition: color 150ms ease-in-out;
  outline: none;

  :hover {
    color: var(--c-text-hover, ${({ theme }) => theme.cAccent});
    cursor: pointer;
  }
`;
export { Anchor };
