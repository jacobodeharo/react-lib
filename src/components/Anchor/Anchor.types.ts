import { ComponentProps } from 'react';
import { Theme } from '../theme/theme.types';

export type AnchorProps = ComponentProps<'a'> & {
  theme?: Theme;
};
