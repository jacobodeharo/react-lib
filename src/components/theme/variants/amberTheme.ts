import { Theme } from '../theme.types';

export const AMBER_THEME: Theme = {
  primaryColorDark: '#FFA000',
  primaryColor: '#FFC107',
  primaryColorLight: '#FFECB3',
  primaryColorText: '#212121',
  accentColor: '#009688',
  primaryTextColor: '#212121',
  secondaryTextColor: '#757575',
  dividerColor: '#BDBDBD',
};
