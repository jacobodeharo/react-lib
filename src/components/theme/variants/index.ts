import { GRAY_THEME } from './grayTheme';
import { AMBER_THEME } from './amberTheme';
import type { Theme } from '../theme.types';

export type ThemeID = 'gray' | 'amber';
export type Themes = {
  [P in ThemeID]: Theme;
};

export const availableThemes: Themes = {
  gray: GRAY_THEME,
  amber: AMBER_THEME,
};
