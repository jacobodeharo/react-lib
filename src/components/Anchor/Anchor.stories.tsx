import React from 'react';
import { AnchorStyles } from './Anchor.styles';
import { AnchorProps } from './Anchor.types';
import { Meta, Story } from '@storybook/react/types-6-0';
import { ThemeProvider } from '../theme';
import { availableThemes } from '../theme/variants';

export default {
  title: 'Components/Base/Anchor',
  component: AnchorStyles,
  argTypes: {
    label: { control: 'text' },
    themeName: {
      control: {
        type: 'select',
        options: Object.keys(availableThemes),
      },
    },
  },
} as Meta;

type StoryButtonProps = AnchorProps & {
  label: string;
  themeName: string;
};

const Template: Story<StoryButtonProps> = (args) => {
  const { label } = args;
  return (
    <ThemeProvider theme={availableThemes[args.themeName]}>
      <AnchorStyles>{label}</AnchorStyles>
    </ThemeProvider>
  );
};

export const Primary = Template.bind({});
Primary.args = {
  label: 'This is an anchor (link)',
  themeName: 'light',
};
