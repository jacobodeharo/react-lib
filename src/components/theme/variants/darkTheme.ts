import { Theme } from '../theme.types';

export const DARK_THEME: Theme = {
  cText: '#e5d3d8',
  cBackground: '#39393c',
  cAccent: '#afcce9',
  cAccentBackground: '#4141a2',
  cStroke: '#e5d3d8',
};

export const lightTheme: Theme = {
  ...DARK_THEME,
};
