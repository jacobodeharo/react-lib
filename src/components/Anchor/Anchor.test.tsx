import * as React from 'react';
import 'jest-styled-components';
import { render, cleanup, screen } from '@testing-library/react';
import { Anchor } from './Anchor';
import { DARK_THEME } from '../theme/variants/darkTheme';

describe('Anchor', () => {
  const ANCHOR_LABEL = 'Anchor Label';

  afterEach(cleanup);

  test('has to be rendered', () => {
    render(
      <Anchor
        href='https://www.test.com'
        rel='noopener noreferer'
        target='_blank'
      >
        {ANCHOR_LABEL}
      </Anchor>
    );
    const anchorElement = screen.getByText(ANCHOR_LABEL).closest('a');
    expect(anchorElement).toMatchSnapshot();
    expect(anchorElement).toHaveAttribute('href', 'https://www.test.com');
    expect(anchorElement).toHaveAttribute('rel', 'noopener noreferer');
    expect(anchorElement).toHaveAttribute('target', '_blank');
  });

  test('has to be rendered with proper styles by using a theme', () => {
    render(
      <Anchor
        href='https://www.test.com'
        rel='noopener noreferer'
        target='_blank'
        theme={DARK_THEME}
      >
        {ANCHOR_LABEL}
      </Anchor>
    );
    const anchorElement = screen.getByText(ANCHOR_LABEL).closest('a');
    expect(anchorElement).toMatchSnapshot();
  });
});
