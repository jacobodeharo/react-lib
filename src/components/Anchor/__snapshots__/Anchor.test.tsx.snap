// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor has to be rendered 1`] = `
.c0 {
  color: var(--c-text,);
  -webkit-transition: color 150ms ease-in-out;
  transition: color 150ms ease-in-out;
  outline: none;
}

.c0:hover {
  color: var(--c-text-hover,);
  cursor: pointer;
}

<a
  class="c0"
  href="https://www.test.com"
  rel="noopener noreferer"
  target="_blank"
>
  Anchor Label
</a>
`;

exports[`Anchor has to be rendered with proper styles by using a theme 1`] = `
.c0 {
  color: var(--c-text,#e5d3d8);
  -webkit-transition: color 150ms ease-in-out;
  transition: color 150ms ease-in-out;
  outline: none;
}

.c0:hover {
  color: var(--c-text-hover,#afcce9);
  cursor: pointer;
}

<a
  class="c0"
  href="https://www.test.com"
  rel="noopener noreferer"
  target="_blank"
>
  Anchor Label
</a>
`;
